src/test/java/org/candlepin/resource/OwnerResourceOwnerClaimingTest.java

```diff
@@ -212,6 +213,31 @@ public void claimantCannotBeAnonymous() {
             .isInstanceOf(BadRequestException.class);
     }
 
+    @Test
+    public void testClaimWhenClaimantExistOnlyUpstream() {
+        Owner originOwner = createAnonOwner(ORIGIN_KEY);
+        when(this.ownerCurator.getByKey(ORIGIN_KEY)).thenReturn(originOwner);
+        when(this.ownerCurator.getByKey(CLAIMANT_KEY)).thenReturn(null);
+        when(this.ownerServiceAdapter.isOwnerKeyValidForCreation(CLAIMANT_KEY)).thenReturn(true);
+        OwnerResource resource = buildOwnerResource();
+
+        resource.claim(ORIGIN_KEY, createClaimant());
+
+        verify(ownerCurator, times(1)).create(any(Owner.class));
+    }
+
+    @Test
+    public void testClaimWhenClaimantDoesNotExistUpstreamOrCandlepin() {
```

Comment:
Add Owner to the test name: `testClaimWhenClaimantOwnerDoesNotExistUpstreamOrCandlepin`

Reply:
done